qa-tests-backend/src/test/groovy/IntegrationsSplunkViolationsTest.groovy

```diff
@@ -119,34 +120,61 @@ class IntegrationsSplunkViolationsTest extends BaseSpecification {
             TimeUnit.SECONDS.sleep(15)
             Response response = SplunkUtil.getSearchResults(port, searchId)
             // We should have at least one violation in the response
-            if (response == null) {
-                continue
-            }
+            assert response != null
             results = response.getBody().jsonPath().getList("results")
-            if (results.isEmpty()) {
-                continue
-            }
+            assert !results.isEmpty()
             hasNetworkViolation = results.any { isNetworkViolation(it) }
             hasProcessViolation = results.any { isProcessViolation(it) }
-            if (hasNetworkViolation && hasProcessViolation) {
-                log.info "Success!"
-                break
-            }
+            log.debug "Found violations in Splunk: \n${results}"
+            log.info "Current Splunk index contains " +
+                    "any Network Violation: ${hasNetworkViolation} and any Process Violation: ${hasProcessViolation}"
+            assert hasNetworkViolation && hasProcessViolation
+        }
+
+        log.info "Starting conversion of ACS violations to Splunk alerts"
+        postToSplunk(port, "/services/saved/searches/" + SPLUNK_TA_CONVERSION_JOB_NAME +"/dispatch", [
+                "dispatch.now": "true",
+                "force_dispatch": "true",
+        ])
+
+        // Check for Alerts
+        List<Map<String, String>> alerts = Collections.emptyList()
+        boolean hasNetworkAlert = false
+        boolean hasProcessAlert = false
+        withRetry(20, 5) {
+            // Hint: If this produces no results, evaluate expanding the earliest_time, e.g. to -15m.
+            // This can be done by expanding `createSearch` with a new parameter.
+            def vSearchId = SplunkUtil.createSearch(port, "| from datamodel Alerts.Alerts")
+            TimeUnit.SECONDS.sleep(15)
+            Response vResponse = SplunkUtil.getSearchResults(port, vSearchId)
+            // We should have at least one violation in the response
+            assert vResponse != null
+            alerts = vResponse.getBody().jsonPath().getList("results")
+            assert !alerts.isEmpty()
+            hasNetworkAlert = alerts.any { isNetworkViolation(it) }
+            hasProcessAlert = alerts.any { isProcessViolation(it) }
+            log.debug "Found Alerts in Splunk: \n${alerts}"
+            log.info "Current Splunk index contains " +
+                    "any Network Alert: ${hasNetworkAlert} and any Process Alert: ${hasProcessAlert}"
+            assert hasNetworkAlert && hasProcessAlert
         }
 
         then:
         "StackRox violations are in Splunk"
         assert !results.isEmpty()
+        assert !alerts.isEmpty()
         assert hasNetworkViolation
         assert hasProcessViolation
-        for (result in results) {
-            validateCimMappings(result)
+        assert hasNetworkAlert
+        assert hasProcessAlert
```

Comment:
If we got to this point I think these assertion are redundant, right?

Reply:
☝️ the diff looks weird in this view. I meant these assertions:

```groovy
        assert !results.isEmpty()
        assert !alerts.isEmpty()
        assert hasNetworkViolation
        assert hasProcessViolation
        assert hasNetworkAlert
        assert hasProcessAlert
```

Reply:
Good point. I will leave in the assertion for alerts not being empty, as we're passing that to `validateCimMapings`, but everything else has been asserted true already. So:
```groovy
        assert !alerts.isEmpty()
	log.info "Validating CIM mappings for alerts"
        for (alert in alerts) {
            validateCimMappings(alert)
	}
```